test/components/componentTests.ts

```diff
@@ -794,20 +794,50 @@ describe("Component", () => {
 
   describe("restricting rendering through clipPath", () => {
 
-    let c: Plottable.Component;
+    let clippedComponent: Plottable.Component;
     let svg: d3.Selection<void>;
 
     beforeEach(() => {
-      c = new Plottable.Component();
+      clippedComponent = new Plottable.Component();
+      (<any> clippedComponent)._clipPathEnabled = true;
       svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
     });
 
     it("generates a clipPath element if it is enabled", () => {
-      (<any> c)._clipPathEnabled = true;
-      c.anchor(svg);
+      clippedComponent.anchor(svg);
       let componentElement = svg.select(".component");
       assert.isNotNull(componentElement.attr("clip-path"), "clip-path attribute set");
-      c.destroy();
+      clippedComponent.destroy();
+      svg.remove();
+    });
+
+    it("uses the correct clipPath", () => {
+      clippedComponent.renderTo(svg);
+      TestMethods.verifyClipPath(clippedComponent);
+      svg.remove();
+    });
+
+    it("updates the clipPath reference when render()-ed", () => {
+      if (window.history && window.history.replaceState) { // not supported on IE9 (http://caniuse.com/#feat=history)
```

Comment:
I prefer a return instead of a huge if block.  1 extra level of nesting isn't fun...